staging/src/k8s.io/apimachinery/pkg/util/wait/wait.go

```diff
@@ -190,19 +216,11 @@ type Backoff struct {
 // If the condition never returns true, ErrWaitTimeout is returned. All other
 // errors terminate immediately.
 func ExponentialBackoff(backoff Backoff, condition ConditionFunc) error {
-	duration := backoff.Duration
-	for i := 0; i < backoff.Steps; i++ {
-		if i != 0 {
-			adjusted := duration
-			if backoff.Jitter > 0.0 {
-				adjusted = Jitter(duration, backoff.Jitter)
-			}
-			time.Sleep(adjusted)
-			duration = time.Duration(float64(duration) * backoff.Factor)
-		}
+	for backoff.Steps > 0 {
 		if ok, err := condition(); err != nil || ok {
 			return err
 		}
+		time.Sleep(backoff.Step())
```

Comment:
doesn't this mutate the passed in backoff object, making it unfit for reuse? a common pattern is to define backoff params in a shared var for reuse (c.f. UpdateTaintBackoff, UpdateNodeSpecBackoff, updateNetworkConditionBackoff, RemoveTokenBackoff, EvictionsRetry, updateBackoff, certificateWaitBackoff, DefaultRetry, DefaultBackoff)

Reply:
nm, function arg is pass-by-value. Step() receiver being a pointer threw me